File: leet_3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        char_set = set() 
        longest = 0 
        l = 0
        for i in range(len(s)):
            while s[i] in char_set:
               char_set.remove(s[l]) 
               l += 1 

            longest = max(longest, (i - l)+1 )
            char_set.add(s[i]) 
            
        return longest 
      
        
            
s = "pwwkew" 

call = Solution()
print(call.lengthOfLongestSubstring(s)) 

File: leet_3.ts
function lengthOfLongestSubstring(s: string): number {
  const char_set = new Set<string>();
  let longest: number = 0;
  let l: number = 0;

  for (let r: number = 0; r < s.length; r++) {
    while (char_set.has(s[r])) {
       char_set.delete(s[l]);
       l++;
    }

    char_set.add(s[r]);
    longest = Math.max(longest, r - l + 1);
  }

  return longest;
}



const s = "pwwkew";

console.log(lengthOfLongestSubstring(s));
